Mage.Sets/src/mage/cards/d/DonalHeraldOfWings.java

```diff
@@ -0,0 +1,145 @@
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.stack.Spell;
+import mage.game.stack.StackAbility;
+import mage.game.stack.StackObject;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+import mage.util.functions.StackObjectCopyApplier;
+import mage.watchers.Watcher;
+
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public class DonalHeraldOfWings extends CardImpl {
+
+    private static final FilterSpell filterSpell = new FilterSpell("a nonlegendary creature spell with flying");
+    static {
+        filterSpell.add(Predicates.not(SuperType.LEGENDARY.getPredicate()));
+        filterSpell.add(CardType.CREATURE.getPredicate());
+    }
+
+    public DonalHeraldOfWings(UUID ownderId, CardSetInfo setInfo) {
+        super(ownderId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{U}{U}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+
+        this.addSubType(SubType.HUMAN);
+        this.addSubType(SubType.WIZARD);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Whenever you cast a nonlegendary creature spell with flying, you may copy it,
+        // except the copy is a 1/1 Spirit in addition to its other types.
+        // Do this only once each turn. (The copy becomes a token.)
+        // TODO: This still triggers and asks if you wanna use it, even if you've used it once this turn.
+        Ability ability = new SpellCastControllerTriggeredAbility(new DonalHeraldOfWingsEffect(), filterSpell, true, true);
+        ability.setIdentifier(MageIdentifier.DonalHeraldOfWingsWatcher);
+        this.addAbility(ability, new DonalHeraldOfWingsWatcher());
+    }
+
+    private DonalHeraldOfWings(final DonalHeraldOfWings card) { super(card); }
+
+    @Override
+    public DonalHeraldOfWings copy() { return new DonalHeraldOfWings(this); }
+}
+
+class DonalHeraldOfWingsEffect extends OneShotEffect {
+
+    DonalHeraldOfWingsEffect() {
+        super(Outcome.Copy);
+        staticText = "you may copy it, except the copy is a 1/1 Spirit in addition to its other types.";
+    }
+
+    private DonalHeraldOfWingsEffect(final DonalHeraldOfWingsEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) { return false; }
+
+        DonalHeraldOfWingsWatcher watcher = game.getState().getWatcher(DonalHeraldOfWingsWatcher.class);
+        if (watcher == null) { return false; }
+        if (watcher.wasAbilityUsedThisTurn()) { return false; }
+
+        // Get the card that was cast
+        if (this.getTargetPointer() == null) { return false; }
+        Spell originalSpell = game.getStack().getSpell(((FixedTarget) this.getTargetPointer()).getTarget());
+
+        // Create a token copy
+        // TODO: Copy made is a token, but there's no indication of it on the card
```

Comment:
The cards enter as tokens (they don't go to the graveyard when they die, but there's no indication on them that they're a token)

Reply:
a copy of a permanent spell enters as a token, it says so on the reminder text

Reply:
Sorry I wasn't clear. What I meant is that Xmage does not seem to give very prominent info about which one is the token. I expected a little icon to show up (e.g. like the ones for flying), or something visual to see.

The only indication is that the card type changed from "permanent" to "token"

Reply:
Oh I thought tokens had an icon, I guess not. @JayDi85 we should probably have a token icon.

Reply:
There are must be special icon for tokens in the top left corner (trace of the animal symbol). Like copy or transformable symbols.